controllers/apps/components/base_stateful.go

```diff
@@ -700,6 +701,18 @@ func (c *statefulComponentBase) updateWorkload(stsObj *appsv1.StatefulSet) bool
 	return false
 }
 
+func (c *statefulComponentBase) updateUpdateStrategy(stsObj, stsProto *appsv1.StatefulSet) {
+	var objMaxUnavailable *intstr.IntOrString
+	if stsObj.Spec.UpdateStrategy.RollingUpdate != nil {
+		objMaxUnavailable = stsObj.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable
+	}
+	stsObj.Spec.UpdateStrategy = stsProto.Spec.UpdateStrategy
+	if objMaxUnavailable == nil && stsObj.Spec.UpdateStrategy.RollingUpdate != nil {
+		// HACK: seems k8s not support it, we set it nil explicitly to avoid the workload been updated unexpectedly.
```

Comment:
Some suggestion to refine the comment. According to k8s api doc  
> This field is alpha-level and is only honored by  servers that enable the MaxUnavailableStatefulSet feature  
 
Suggest to add more infor about this feature. To avoid others being trapped.

Reply:
update